File: src/main/kotlin/com/stannis/dataModel/statementTypes/CompositeTypeSpecifier.kt
package com.stannis.dataModel.statementTypes

import com.stannis.dataModel.DeclarationParent
import com.stannis.dataModel.DeclarationSpecifierParent
import com.stannis.dataModel.NameInterface
import com.stannis.dataModel.Statement

data class CompositeTypeSpecifier(
    override val `$type`: String? = "CompositeTypeSpecifier",
    var name: NameInterface?,
    var baseSpecifier: ArrayList<BaseSpecifier>?,
    var declarations: ArrayList<DeclarationParent>?,
    var key: Int?
) : Statement, DeclarationSpecifierParent {
    fun addStatement(statement: Statement) {
        if (declarations == null) {
            declarations = ArrayList()
        }
        declarations!!.add(statement as DeclarationParent)
    }

    fun addBase(statement: Statement) {
        if (baseSpecifier == null) {
            baseSpecifier = ArrayList()
        }
        baseSpecifier!!.add(statement as BaseSpecifier)
    }
}

File: Interpreter/Interpreter/Models/serialize/statementTypes/CompositeTypeSpecifier.cs
using System.Collections.Generic;

namespace Interpreter.Models.serialize.statementTypes
{
    public class CompositeTypeSpecifier : IStatement
    {
        public IStatement name { set; get; }

        public IList<IStatement> baseSpecifier = new List<IStatement>();

        public IList<IStatement> declarations = new List<IStatement>();

        public int key { set; get; }

    }
}